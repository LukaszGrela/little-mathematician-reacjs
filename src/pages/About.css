.about {
    padding-top: 1em;
    text-align: left;
    width: 300px;
    margin: 0 auto;
}
.about .title {
    font-size: 2em;
    line-height: 3em;
}
.about .acknowledgement {
    color: #757575;
}