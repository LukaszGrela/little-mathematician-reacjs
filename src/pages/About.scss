/*
   Copyright 2018 ≈Åukasz 'Severiaan' Grela GrelaDesign

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
*/

@import '../sass/_variables.scss';
.about {
    padding: 1em;
    text-align: left;
    width: $width-game-column;
    margin: 0 auto;
    .title {
        font-size: 2em;
        line-height: 3em;
    }
    .acknowledgement {
        color: $secondary-text-color;
    }
    .git-source {
        padding: 1rem 0;
        .git-hub-icon {
            vertical-align: bottom;
            width: 1.4rem;
            height: 1.4rem;
            padding-right: 0.4rem;
        }
    }
    
}