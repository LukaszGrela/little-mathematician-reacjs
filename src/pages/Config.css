.config {
    padding-top: 50px;
}
.config label {
    display: block;
    padding-bottom: 8px;
}
.config input,
.config select {
    margin-left: 8px;
}