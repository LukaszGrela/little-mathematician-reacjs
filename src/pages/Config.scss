

.config {
    padding-top: 50px;
    label {
        display: block;
        padding-bottom: 8px;
    }
    input,
    select {
        margin-left: 8px;
    }
}
