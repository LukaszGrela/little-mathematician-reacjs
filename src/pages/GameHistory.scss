.game-history {
    &-row {
      display: flex;
      justify-content: center;
      align-items: center;
      & .score {
        padding: 0 1rem;
      }
    }
}