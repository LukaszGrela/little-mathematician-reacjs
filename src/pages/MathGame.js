/*
   Copyright 2018 Łukasz 'Severiaan' Grela GrelaDesign

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
*/

import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { connect } from 'react-redux';
import GameHud from '../components/GameHud'
import GameAnswers from '../components/GameAnswers'
import Equation from '../components/Equation'
import GameOver from '../components/GameOver'
import Feedback from '../components/Feedback'

import './MathGame.scss';
import { newGame, quitGame, answerQuestion, nextQuestion, gameOver } from '../actions/mathGameActions';

export class MathGame extends Component {

    componentDidMount() {
        this.props.newGame(this.props.config);
    }

    gameOverActionHandler = (action) => {

        const { hudQuestionCurrent, questionCount } = this.props.game;
        this.props.gameOver();
    }
    feedbackActionHandler = () => {
        this.props.nextQuestion();
    }
    gameAnswerSelectionHandler = (option, id) => {
        this.props.answerQuestion(option, id);
    }





    gameOverView = () => {
        const { hudCorrectAnswers } = this.props.game;
        return (
            <GameOver onAction={this.gameOverActionHandler}
                hudCorrectAnswers={hudCorrectAnswers} />
        )
    }
    getNextEquationView = () => {
        const { type, game } = this.props;
        const { operation, questions, hudQuestionCurrent } = game;
        const current = questions[hudQuestionCurrent - 1];
        return <Equation
            {...current}
            operation={operation} />;
    }
    gameView = () => {
        const { type, game } = this.props;
        const { hudCorrectAnswers, hudQuestionCurrent, questionCount, questions } = game;
        const current = questions[hudQuestionCurrent - 1];
        console.log('current', current);
        return (
            <div className='game-view'>
                <GameHud
                    hudCorrectAnswers={hudCorrectAnswers}
                    hudQuestionCurrent={hudQuestionCurrent}
                    questionCount={questionCount}
                    type={type} />
                {
                    this.getNextEquationView()
                }
                {
                    /* show feedback panel if question has user answer */
                    current && current.answer ?
                        <Feedback {...current}
                            onAction={this.feedbackActionHandler} /> : null
                }
                <GameAnswers
                    selectionId={current.answer ? current.answer.selectionId : null}
                    options={current.distractors}
                    onSelection={this.gameAnswerSelectionHandler} />
            </div>
        );
    }

    render() {
        const { type, game } = this.props;
        console.log(game);
        if (game.hasOwnProperty('questionCount') === false) return null;
        const { hudQuestionCurrent, questionCount } = game;
        const isGameOver = hudQuestionCurrent > questionCount;
        return (
            <div className={'game game-' + type.split(':').join('')}>
                {
                    isGameOver ? this.gameOverView() : this.gameView()
                }
            </div>
        );
    }
}

MathGame.propTypes = {
    type: PropTypes.string.isRequired,
    newGame: PropTypes.func.isRequired,
    quitGame: PropTypes.func.isRequired,
    answerQuestion: PropTypes.func.isRequired,
    nextQuestion: PropTypes.func.isRequired,
    gameOver: PropTypes.func.isRequired,
}

const mapStateToProps = (state, props) => ({
    config: { ...state.config, type: props.type },
    game: { ...state.game.currentGame }
});

const mapDispatchToProps = (dispatch) => ({
    newGame: (config) => dispatch(newGame(config)),
    quitGame: () => dispatch(quitGame()),
    answerQuestion: (answer, optionId) => dispatch(answerQuestion(answer, optionId)),
    nextQuestion: () => dispatch(nextQuestion()),
    gameOver: () => dispatch(gameOver()),
});

export default connect(mapStateToProps, mapDispatchToProps)(MathGame);