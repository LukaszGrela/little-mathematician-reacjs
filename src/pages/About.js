/*
   Copyright 2018 ≈Åukasz 'Severiaan' Grela GrelaDesign

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
*/

import React, { Component } from 'react';

import './About.scss'
import GitHubIcon from '../icons/GitHubIcon';

class About extends Component {
    state = {}
    render() {
        return (
            <div className='about'>
                <div className='title'>Little Mathematician</div>
                <div className='content'>
                    Simple Maths game for children made in ReactJS with aim to explore this framework.
                </div>
                <div className='git-source'>
                    <GitHubIcon/><a href="https://github.com/LukaszGrela/little-mathematician-reacjs" 
                    target='_blank' 
                    rel="noopener noreferrer">LukaszGrela/little-mathematician-reacjs</a>
                </div>
                <div className='acknowledgement'>
                    <ul>
                        <li>Icons in menu (and logo) by Leona Grande from the Noun Project, thenounproject.com/</li>
                        <li>Other icons came from material ui icons, www.materialui.co/</li>
                    </ul>
                </div>
            </div>
        );
    }
}

export default About;
