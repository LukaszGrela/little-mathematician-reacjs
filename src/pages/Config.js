/*
   Copyright 2018 Łukasz 'Severiaan' Grela GrelaDesign

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
*/

import React, { Component } from 'react';

import './Config.css'
import connect from 'react-redux/lib/connect/connect';
import { changeNumberOfQuestions,ConfigType, changeRangeTo, changeRangeFrom } from '../actions/actions';

class Config extends Component {


    constructor(props) {
        super(props);
        this.state = Object.assign({}, props);
    }



    handleChange(field, value) {
        let _val = parseInt(value, 10);
        let gCount = this.props.general.questionCount;
        let gFrom = this.props.general.from;
        let gTo = this.props.general.to;
        const { dispatch } = this.props;
        switch (field) {
            case 'general-qn':
                dispatch(changeNumberOfQuestions(_val, ConfigType.GENERAL));
                break;
            case 'general-rf':
                if (_val >= gTo) {
                    dispatch(changeRangeTo(_val + 10, ConfigType.GENERAL));
                }
                //this.setState({ general: { from: _val, to: gTo, questionCount: gCount } });
                dispatch(changeRangeFrom(_val, ConfigType.GENERAL));
                break;
            case 'general-rt':
                if (_val <= gFrom) {
                    dispatch(changeRangeFrom(_val - 10, ConfigType.GENERAL));
                }
                //this.setState({ general: { to: _val, from: gFrom, questionCount: gCount } });
                dispatch(changeRangeTo(_val, ConfigType.GENERAL));
                break;
            default:
                break;
        }
    }


    render() {
        return (
            <div className='config'>
                <form>
                    <div className='general'>
                        <label>
                            Number of questions:
                            <select value={this.props.general.questionCount}
                                onChange={(e) => this.handleChange('general-qn', e.target.value)}>
                                <option value="10">10</option>
                                <option value="25">25</option>
                                <option value="50">50</option>
                                <option value="100">100</option>
                            </select>
                        </label>
                        <label>
                            Number range from:
                            <select value={this.props.general.from}
                                onChange={(e) => this.handleChange('general-rf', e.target.value)}>
                                <option value="0">0</option>
                                <option value="10">10</option>
                                <option value="20">20</option>
                                <option value="30">30</option>
                                <option value="40">40</option>
                            </select>
                        </label>
                        <label>
                            Number range to:
                            <select value={this.props.general.to}
                                onChange={(e) => this.handleChange('general-rt', e.target.value)}>
                                <option value="10">10</option>
                                <option value="20">20</option>
                                <option value="30">30</option>
                                <option value="40">40</option>
                                <option value="50">50</option>
                            </select>
                        </label>
                    </div>
                    <div className='addition'></div>
                    <div className='subtraction'></div>
                    <div className='multiplication'></div>
                    <div className='division'></div>
                </form>
            </div >
        );
    }
}

const mapStateToProps = (state) => {
    console.log('Config#mapStateToProps', state);
    const config = Object.assign({}, state.config);
    return config;
}

export default connect(mapStateToProps)(Config);