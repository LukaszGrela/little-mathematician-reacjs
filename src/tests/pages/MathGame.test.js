import React from 'react';
import { shallow } from 'enzyme';
import { MathGame } from '../../pages/MathGame';
import { GAME_ADDITION } from '../../gameTypes';

let wrapper, newGame;

beforeEach(() => {
    newGame = jest.fn();
    wrapper = shallow(<MathGame
        type={GAME_ADDITION}
        newGame={newGame}
    />);

})

test('Should render MathGame correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('Should call the newGame method', () => {
    expect(newGame).toHaveBeenCalled();
})
// test('', () => { });