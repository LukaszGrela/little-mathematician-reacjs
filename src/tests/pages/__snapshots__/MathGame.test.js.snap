// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render MathGame correctly 1`] = `
<div>
  MathGame
</div>
`;
